import "mapreduce/yt/interface/protos/extension.proto";

package NImages.NIndex;
option optimize_for = SPEED;

// сокращённая версия TTextProto из cv/imageproc/ocr/recognizer_results/external_formats/yimages/protos/ocrresults.proto
message TTextProtoReducedPB {
    repeated string Texts = 1;
    optional uint32 LangName = 2;
}

message TImageDescriptorPB {
    optional string ImageUrl = 1;       // URL оригинальной картинки
    optional string DocId = 2;          // ID метадокумента
    optional string ThumbId = 3;        // ID тумбы
    optional uint32 Height = 4;         // высота оригинальной картинки в пикселях
    optional uint32 Width = 5;          // ширина оригинальной картинки в пикселях
    optional float VisualQuality = 6;
    optional float ClickRank = 7;
    optional float OcrAreaPercent = 8;
    // 9 is skipped for compatibility
    optional string DocIdBase64 = 10;
    optional string DocImageUrl = 11;
    repeated TTextProtoReducedPB Recognitions = 12;

    // for NItdItp::TImageFilter::AllowedImageSources
    optional string ImageSource = 13 [default = "itditp_images"];
    optional bool IsMainContent = 14;
    
    // for Zen
    optional string MainColor = 15;

    optional uint32 SelectedColor = 16;         // actually ui8
}

message TImageDescriptorsPB {
    repeated TImageDescriptorPB ImageDescriptors = 1;
}

message TImages4PagesPB {
    optional string PageUrl = 1                            [ (NYT.key_column_name) = "PageUrl" ];
    optional TImageDescriptorsPB Images = 2                [ (NYT.column_name) = "Images" ];
    optional TImageDescriptorPB MainImageCandidate = 3     [ (NYT.column_name) = "MainImageCandidate" ];
}